src/server/game/Entities/Player/Player.cpp

```diff
@@ -1937,7 +1937,10 @@ bool Player::TeleportTo(uint32 mapid, float x, float y, float z, float orientati
         {
             Position oldPos = GetPosition();
             if (HasUnitMovementFlag(MOVEMENTFLAG_HOVER))
-                z += GetFloatValue(UNIT_FIELD_HOVERHEIGHT);
+            {
+                float ground = GetMap()->GetHeight(GetPhaseMask(), x, y, z);
```

Comment:
this is completely wrong, teleport destination can be anywhere, including high in the air